Core/automation/lib/python/core/date.py

```diff
@@ -196,49 +264,88 @@ def to_python_datetime(value):
+        .. code-block::
+
+            joda_time = to_joda_datetime(items["date_item"])
 
     Args:
-        value: any known DateTime value.
-    
+        value: Any known DateTime value.
+
     Returns:
         | An ``org.joda.time.DateTime`` representing ``value``.
         | If ``value`` does not have timezone information, the system default
           will be used.
-    
+
     Raises:
-        TypeError: type of ``value`` is not recognized by this package.
+        TypeError: If type of ``value`` is not recognized by this package.
     """
-    if isinstance(value, DateTime):
-            return value
-    
+    if isinstance(value, jodaDateTime):
+        return value
+
     value_zoneddatetime = to_java_zoneddatetime(value)
-    return DateTime(
-        value_zoneddatetime.toInstant(),
-        DateTimeZone.forID(value_zoneddatetime.getZone().getId())
+    return jodaDateTime(
+        value_zoneddatetime.toInstant().toEpochMilli(),
+        jodaDateTimeZone.forID(value_zoneddatetime.getZone().getId())
     )
 
-class pythonTimezone(datetime.tzinfo):
-    """Python tzinfo with ``offset`` in minutes and name ``name``.
-        
+def to_java_calendar(value):
+    """Converts any known DateTime type to a ``java.util.Calendar`` type.
+
+    Examples:
+        .. code-block::
+
+            calendar_time = to_java_calendar(items["date_item"])
+
     Args:
-        offset (int): timezone offset from UTC in minutes.
-        name (str): display name of this instance.
-    """
-    
-    def __init__(self, offset=0, name=""):
-        self.__offset = offset
-        self.__name = name
+        value: Any known DateTime value.
 
-    def utcoffset(self, value):
-        return datetime.timedelta(minutes = self.__offset)
+    Returns:
+        | A ``java.util.Calendar`` representing ``value``.
+        | If ``value`` does not have timezone information, the system default
+          will be used.
 
-    def tzname(self, value):
-        return self.__name
+    Raises:
+        TypeError: If type of ``value`` is not recognized by this package.
+    """
+    if isinstance(value, utilCalendar):
+        return value
 
-    def dst(self, value):
-        return datetime.timedelta(0)
+    value_zoneddatetime = to_java_zoneddatetime(value)
+    new_calendar = utilCalendar.getInstance(utilTimeZone.getTimeZone(value_zoneddatetime.getZone().getId()))
+    new_calendar.set(utilCalendar.YEAR, value_zoneddatetime.getYear())
+    new_calendar.set(utilCalendar.MONTH, value_zoneddatetime.getMonthValue() - 1)
+    new_calendar.set(utilCalendar.DAY_OF_MONTH, value_zoneddatetime.getDayOfMonth())
+    new_calendar.set(utilCalendar.HOUR_OF_DAY, value_zoneddatetime.getHour())
+    new_calendar.set(utilCalendar.MINUTE, value_zoneddatetime.getMinute())
+    new_calendar.set(utilCalendar.SECOND, value_zoneddatetime.getSecond())
+    new_calendar.set(utilCalendar.MILLISECOND, int(value_zoneddatetime.getNano() / 1000000))
+    return new_calendar
 
 # aliases
```

Comment:
I don't see a need for the aliases and think they could cause confusion. I realize these weren't added in this PR, but since you're in there...